src/Temporalio/Client/TemporalConnectionOptions.cs

```diff
@@ -103,6 +114,11 @@ public TemporalConnectionOptions()
         /// </remarks>
         public TemporalRuntime? Runtime { get; set; }
 
+        /// <summary>
+        /// Gets a value indicating whether TLS was explicitly set (even to null).
+        /// </summary>
+        internal bool TlsExplicitlySet { get; private set; }
```

Comment:
I don't think we should take this approach. The difference between a user manually doing `= null` and the default of `null` would be very confusing. I think for .NET we should add a `bool Disabled` property to `TlsOptions` for users that want to explicitly disable TLS. `null` should be considered unset/default (regardless of whether it's set that way)

Reply:
Yeah, this is a bit nicer. Done